pkg/apis/core/validation/validation.go

```diff
@@ -3335,6 +3378,11 @@ func validateContainers(containers []core.Container, volumes map[string]core.Vol
 		if ctr.StartupProbe != nil && ctr.StartupProbe.SuccessThreshold != 1 {
 			allErrs = append(allErrs, field.Invalid(path.Child("startupProbe", "successThreshold"), ctr.StartupProbe.SuccessThreshold, "must be 1"))
 		}
+
+		// These fields are disallowed for regular containers
+		if ctr.RestartPolicy != nil {
+			allErrs = append(allErrs, field.Forbidden(path.Child("restartPolicy"), "may not be set for a regular container"))
```

Comment:
Generally we don't say `regular container`, we would just say `container` (especially since the field error has a path that includes `spec.containers.` as a prefix.